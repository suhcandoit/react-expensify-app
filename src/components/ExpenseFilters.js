import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseFilters extends React.Component {
    state = {
        calendarFocus: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocus) => {
        this.setState(()=> ({ calendarFocus }))
    };
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSelectChange = (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate()
        } else if (e.target.value === 'amount') {
            this.props.sortByAmount()
        }               
    } 
    render(){
        return(
            <div>
            <input 
                type='text' 
                value={this.props.filters.text}
                onChange={this.onTextChange}/>
            <select onChange={this.onSelectChange}>
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>

            <DateRangePicker
                startDate={this.props.filters.startDate} 
                endDate={this.props.filters.endDate}  
                onDatesChange={this.onDatesChange}  
                focusedInput={this.state.calendarFocus}  
                onFocusChange={this.onFocusChange}  
                numberOfMonths= {1}
                isOutsideRange={()=>false}
                showClearDates={true}                
            />
        </div>    
        );
    }
}

//setup value and onChange for select
const mapStateToProps = (state) => ({ filters: state.filters })

const mapDispatchToProps = (dispatch) => ({
    setTextFilter:(data) => dispatch(setTextFilter(data)),
    setStartDate:(data) => dispatch(setStartDate(data)),
    setEndDate:(data) => dispatch(setEndDate(data)),
    sortByDate:() => dispatch(sortByDate()),
    sortByAmount:() => dispatch(sortByAmount())
})
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseFilters);