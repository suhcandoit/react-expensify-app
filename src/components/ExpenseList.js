import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import visibleExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.length===0 ?
            (<p>No expenses!</p>) : 
            (props.expenses.map((expense, key) => {
                return <ExpenseListItem key={key} {...expense} />})
        )}
    
    </div>
);

const mapStateToProp = (state) => {
    return {
        expenses:visibleExpenses(state.expenses, state.filters)
    };
}

export default connect(mapStateToProp)(ExpenseList);

// const ConnectedExpenseList = connect((state) => {
//     return {
//         expenses:state.expenses
//     };
// })(ExpenseList);
