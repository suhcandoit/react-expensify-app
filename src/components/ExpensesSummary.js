import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import visibleExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => {
    return(
        <div>
            {<h1>Viewing {props.expenses.length} totalling {numeral(selectExpensesTotal(props.expenses)/100).format('$0,0.00')}</h1>}
        </div>)
}

const mapStateToProp = (state) => {
    return({
        expenses: visibleExpenses(state.expenses, state.filters)
    })
}

export default connect(mapStateToProp)(ExpensesSummary);
