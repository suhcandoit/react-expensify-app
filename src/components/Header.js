import React from 'react';
import { NavLink } from 'react-router-dom'; 
import { connect } from 'react-redux';
import { startLogOut } from '../actions/auth';

export const Header = ({ startLogOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink exact to="/" activeClassName='selected'>Home</NavLink>
        <NavLink to="/create" activeClassName='selected'>Add Expense</NavLink>
        <NavLink to="/help" activeClassName='selected'>Help Page</NavLink>
        <button onClick={ startLogOut }>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
});
  
export default connect(undefined, mapDispatchToProps)(Header);

