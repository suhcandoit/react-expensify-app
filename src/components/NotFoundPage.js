import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    // It uses to link internally without refresh/reloading
    <div> 
        404 Page - <Link to='/'>Go home</Link>
    </div>
);

export default NotFoundPage;


