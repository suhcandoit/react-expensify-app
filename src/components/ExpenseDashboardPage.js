import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseFilters from './ExpenseFilters';

const ExpenseDashboardPage = () => (
    <div>Expense Dashboard
        <ExpenseFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;