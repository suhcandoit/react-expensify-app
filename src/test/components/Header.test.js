import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('Should render Header correctly', () => {
    const wrapper = shallow(<Header startLogOut={() => {}} />);
    expect(wrapper).toMatchSnapshot()
})

test('should call startLogout on button click', () => {
    const startLogOut = jest.fn();
    const wrapper = shallow(<Header startLogOut={startLogOut} />);
    wrapper.find('button').simulate('click');
    expect(startLogOut).toHaveBeenCalled();   
})

// LoginPage test file -> Should call startLogin on button click
