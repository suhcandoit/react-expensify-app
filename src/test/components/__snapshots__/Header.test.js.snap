// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="selected"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="selected"
    to="/create"
  >
    Add Expense
  </NavLink>
  <NavLink
    activeClassName="selected"
    to="/help"
  >
    Help Page
  </NavLink>
</header>
`;
